pkg/lib/oauth/handler/handler_token.go

```diff
@@ -585,7 +585,11 @@ func (h *TokenHandler) IssueTokensForAuthorizationCode(
 	}
 
 	dpopProof := dpop.GetDPoPProof(ctx)
-	if !codeGrant.MatchDPoPJKT(dpopProof) {
+	if err := codeGrant.MatchDPoPJKT(dpopProof); err != nil {
+		logger.WithSkipLogging().Error(ctx,
+			fmt.Sprintf("failed to match dpop jkt on issue tokens: %s", err.Message),
+			err.Info_ReadOnly.ToSlogAttrs()...,
+		)
```

Comment:
With my suggestion implemented,

1. We no longer need to call `err.Info_ReadOnly.ToSlogAttrs()...`. The handling of ToSlogAttrs should be handled by slogutil behind the scene. I suggest we add a new handler to do this. This handler has to come before the mask handler. Note that the handler has to use errorutil.Unwrap to traverse the error chain and collect all ToSlogAttrs(). errorutil.Unwrap is smart enough to handle all kinds of error wrapping.
2. Use WithError to include the error.

So this code should look like

```golang
// Assume the slog setup supports ToSlogAttrs for errors, including wrapped ones.
logger.WithSkipLogging().WithError(err).Error(ctx, "failed to match dpop jkt on issue tokens")
```